pkg/wrapper/buffered_tree.go

```diff
@@ -0,0 +1,188 @@
+package wrapper
+
+import (
+	"fmt"
+	"runtime"
+
+	"github.com/celestiaorg/celestia-app/v6/pkg/appconsts"
+	"github.com/celestiaorg/go-square/v3/share"
+	"github.com/celestiaorg/nmt"
+	"github.com/celestiaorg/rsmt2d"
+)
+
+// TreePool provides a fixed-size pool of resizeableBufferTree instances.
+type TreePool struct {
+	availableNMTs chan *resizeableBufferTree
+	opts          []nmt.Option
+	poolSize      int
+}
+
+// DefaultPreallocatedTreePool creates a new TreePool with a default pool size
+func DefaultPreallocatedTreePool(squareSize uint) *TreePool {
+	return NewTreePool(squareSize, runtime.NumCPU()*4)
+}
+
+// NewTreePool creates a new TreePool with the specified configuration.
+func NewTreePool(initSquareSize uint, poolSize int, opts ...nmt.Option) *TreePool {
+	pool := &TreePool{
```

Comment:
does it make sense to add a zero pool size case where we return an error or set a default? Just to foolproof it

Reply:
Yes, we can probably just use a default value for that for simplicity